src/server/ocpp/utils/OCPPUtils.ts

```diff
@@ -692,6 +692,24 @@ export default class OCPPUtils {
 
   public static updateTransactionWithStopTransaction(transaction: Transaction, stopTransaction: OCPPStopTransactionRequestExtended,
       user: User, alternateUser: User, tagId: string): void {
+    // Handle Signed Data
+    for(const meterValue of (stopTransaction.transactionData as OCPPMeterValue[])){
+      if(meterValue.sampledValue){
+        for(const sampledValue of meterValue.sampledValue){
+          if (sampledValue.format === OCPPValueFormat.SIGNED_DATA) {
+            if (sampledValue.context === OCPPReadingContext.TRANSACTION_BEGIN) {
+              // Set the first Signed Data and keep it
+              transaction.signedData = sampledValue.value as string;
+              continue;
+            } else if (sampledValue.context === OCPPReadingContext.TRANSACTION_END) {
+              // Set the last Signed Data (used in the last consumption)
+              transaction.currentSignedData = sampledValue.value as string;
+              continue;
```

Comment:
I do not understand why you need to specify a continue after each findings. 

And that code duplicate a similar code in `updateTransactionWithMeterValues()`. Please factor out the code digging the metervalues.

Reply:
the code is factored out in OCPPUtils